nixos/modules/services/networking/hickory-dns.nix

```diff
@@ -126,23 +135,23 @@ in
   };
 
   config = lib.mkIf cfg.enable {
-    systemd.services.trust-dns = {
-      description = "trust-dns Domain Name Server";
-      unitConfig.Documentation = "https://trust-dns.org/";
+    systemd.services.hickory-dns = {
+      description = "hickory-dns Domain Name Server";
+      unitConfig.Documentation = "https://hickory-dns.org/";
       serviceConfig = {
         ExecStart =
         let
           flags =  (lib.optional cfg.debug "--debug") ++ (lib.optional cfg.quiet "--quiet");
           flagsStr = builtins.concatStringsSep " " flags;
         in ''
-          ${cfg.package}/bin/trust-dns --config ${configFile} ${flagsStr}
+          ${cfg.package}/bin/hickory-dns --config ${configFile} ${flagsStr}
```

Comment:
without bumping the package version, this change doesn't work. 0.23.0 still ships only `bin/trust-dns` and not `bin/hickory-dns`. and 0.24.0 ships `bin/hickory-dns` only, without `bin/trust-dns`. so unless we want to introduce and leverage `meta.mainProgram`, there has to be a patch which simultaneously upgrades 0.23.0 -> 0.24.0 and changes the name here.

Reply:
here's a PR which, if merged before the rebrand, would address this issue: <https://github.com/NixOS/nixpkgs/pull/263054>

Reply:
PR above has been merged, should be safe to just leave this line as it's set in master